tensorflow/tools/ci_build/builds/run_pip_tests.sh

```diff
@@ -122,14 +140,21 @@ else
 fi
 
 export TF_NEED_CUDA=$IS_GPU
-${PYTHON_BIN_PATH} configure.py
+export TF_NEED_ROCM=$IS_ROCM
+yes "" | ${PYTHON_BIN_PATH} configure.py
```

Comment:
Why is "yes" needed here? That doesn't look ROCM-specific

Reply:
The `configure.py` script requires user input...not having the `yes "" | ` preceding it, will make the `run_pip_tests.sh` at that command

```
...
root@ixt-rack-04:/root/tensorflow# /usr/bin/python3 configure.py 
You have bazel 3.1.0 installed.
Please specify the location of python. [Default is /usr/bin/python3]: 
...
```

The `yes ""` command simply picks the default options for all user input

Reply:
Can we instead populate `.tf_configure.bazelrc` and not run `configure.py` at all?

Reply:
hmm....think we are not in sync.

I thought `.tf_configure.bazelrc` is file created+written to, when we run the `configure.py` script.  

If the end result is updating the `.tf_configure.bazelrc` file, does it matter whether we modify it directly or via the `configure.py` script?

Reply:
> I thought .tf_configure.bazelrc is file created+written to, when we run the configure.py script.

Yes.

> If the end result is updating the .tf_configure.bazelrc file, does it matter whether we modify it directly or via the configure.py script?

Piping `yes` to a configuration script might have unexpected results: it's really not clear what is it doing, and it's subject to changes if `configure.py` changes. I think it's better to explicitly provide a configuration file instead.

Reply:
@sanjoy

Reply:
I agree that checking in a pre-populated `.tf_configure.bazelrc` is better.  I believe @av8ramit tried fixing this for the other build scripts we have, which had to be rolled back.  Amit, do you remember why it had to be rolled back?

Reply:
@pak-laura tried to remove the yes pipe logic, but it was deemed infeasible due to issues with bazel. Laura, any chance you can reply here with a summary of the issues you faced.

Reply:
+1 to having configuration checked in. We were recently hit by `--march-native` being written by default in `.tf_configure.bazelrc` which made TF 2.4 wheels require AVX2.

Reply:
I am not quite sure I understand what the  "pre-populated `.tf_configure.bazelrc` file" solution should look like.

Currently the generated `.tf_configure.bazelrc` file looks like
```
build --action_env PYTHON_BIN_PATH="/usr/bin/python3"                                
build --action_env PYTHON_LIB_PATH="/usr/local/lib/python3.6/dist-packages"          
build --python_path="/usr/bin/python3"                                               
build --config=xla                                                                   
build --config=rocm                                                                  
build --action_env ROCM_PATH="/opt/rocm-4.0.0"                                       
build:opt --copt=-march=native                                                       
build:opt --copt=-Wno-sign-compare                                                   
build:opt --host_copt=-march=native                                                  
build:opt --define with_default_optimizations=true                                   
test --flaky_test_attempts=3                                                         
test --test_size_filters=small,medium                                                
test --test_env=LD_LIBRARY_PATH                                                      
test:v1 --test_tag_filters=-benchmark-test,-no_oss,-no_gpu,-oss_serial               
test:v1 --build_tag_filters=-benchmark-test,-no_oss,-no_gpu                          
test:v2 --test_tag_filters=-benchmark-test,-no_oss,-no_gpu,-oss_serial,-v1only       
test:v2 --build_tag_filters=-benchmark-test,-no_oss,-no_gpu,-v1only                  
build --action_env TF_CONFIGURE_IOS="0"                                              
```

It contains options within it that are
* applicable to all (for eg.  `--config=xla` )
* GPU specific (for eg. `--test_tag_filters`)
* ROCm specific (for eg `--config=rocm` )
* env specific (for eg `--action_env PYTHON_BIN_PATH=...`)


What would a "checked-in" version of `.tf_configure.bazelrc` file look like? 

My understanding was that options that are not specific to a given "configuration" are already a part of the `.bazelrc` file in the TF repo, and that most (if not all of) the contents of the `.tf_configure.bazelrc` are dependent on the user specified build parameters (python location, ROCm platform, etc). Given this, 
* how would one determine what should go into a checked-in / pre-populated version of the `.tf_configure.bazelrc` file?
* we will still need to modify that file at "runtime" because some options like the ROCM_PATH are user env specific

Reply:
@deven-amd Good point, I didn't consider that `configure.py` needs to find some paths. Considering we already have yes-piping in the repo in a few places, this should be OK (or a separate effort needs to be made to remove them all)